core/domain/stats_services.py

```diff
@@ -646,7 +646,10 @@ def delete_playthroughs_multi(playthrough_ids):
     Args:
         playthrough_ids: list(str). List of playthrough IDs to be deleted.
     """
-    stats_models.PlaythroughModel.delete_multi(
+    # Run in transaction to help prevent data-races between concurrent
+    # operations that may update the playthroughs being deleted.
+    transaction_services.run_in_transaction(
+        stats_models.PlaythroughModel.delete_multi,
         stats_models.PlaythroughModel.get_multi(playthrough_ids))
```

Comment:
Shouldn't we transact the gets & deletes together? At the moment, this is running get_multi outside the transaction. I think it's still possible here for a playthrough to be deleted between get_multi and delete_multi.

Reply:
Ah yes, great catch! Done.